c++/include/orc/MemoryPool.hh

```diff
@@ -51,6 +51,7 @@ namespace orc {
 
    public:
     DataBuffer(MemoryPool& pool, uint64_t _size = 0);
+    DataBuffer(MemoryPool& pool, uint64_t _size, bool noMemSet);
```

Comment:
When do we use `noMemSet=false`?

Reply:
No. The problem is that it is possible to provide explicit `template <bool noMemSet>` to a constructor. Perhaps we can allow this branch in constructor? That makes the code better but introduce a branch that should be resolved at compile time.

Reply:
In the new commit I tried to use an empty class tag type to differentiate the constructor